### PR TITLE
lsa checks and alpc info

### PR DESCRIPTION
**Pull request by jxy-s**

### CODE REVIEW

File: KSystemInformer/util.c [hunk 1]
```diff
@@ -1472,3 +1477,247 @@ NTSTATUS KphGetThreadExitStatus(
...
+    NTSTATUS status;
+    HANDLE portHandle;
+    KAPC_STATE apcState;
+    KPH_ALPC_COMMUNICATION_INFORMATION info;
+
+    PAGED_PASSIVE();
+
+    *ProcessId = NULL;
+
+    KeStackAttachProcess(PsInitialSystemProcess, &apcState);
```
Comment (jxy-s) [inline KSystemInformer/util.c:1598]: Add comments here

File: KSystemInformer/util.c [hunk 2]
```diff
@@ -1472,3 +1477,247 @@ NTSTATUS KphGetThreadExitStatus(
...
+
+/**
+ * \brief Retrieves the process ID of lsass from the registry. 
+ *
+ * \param[out] ProcessId Set to the process ID of lsass.
+ *
+ * \return Successful or errant status.
+ */
+_IRQL_requires_max_(PASSIVE_LEVEL)
+NTSTATUS KphpGetLsassProcessIdReg(
```
Comment (jxy-s) [inline KSystemInformer/util.c:1523]: Remove reg path, ALPC is da way.

File: KSystemInformer/util.c [hunk 3]
```diff
@@ -89,6 +89,11 @@ VOID KphReleaseRundown(
 
 PAGED_FILE();
 
+static UNICODE_STRING KphpLsaPortName = RTL_CONSTANT_STRING(L"\\SeLsaCommandPort");
+static UNICODE_STRING KphpLsaKey = RTL_CONSTANT_STRING(L"\\Registry\\Machine\\System\\CurrentControlSet\\Control\\Lsa");
```
Comment (jxy-s) [inline KSystemInformer/util.c:93]: To be removed.

File: KSystemInformer/cid_tracking.c [hunk 4]
```diff
@@ -186,6 +188,13 @@ NTSTATUS KSIAPI KphpInitializeProcessContext(
         {
             process->VerifiedProcess = TRUE;
         }
+
+        if (!KphpLsassIsKnown &&
```
Comment (jxy-s) [inline KSystemInformer/cid_tracking.c:192]: @ionescu007 what are the chances multiple lsass processes at once are important to handle?

**Review by dmex — APPROVED** [review]

**Pull request merged**